cordova-lib/src/cordova/plugin.js

```diff
@@ -283,6 +283,31 @@ module.exports = function plugin(command, targets, opts) {
     }
 };
 
+/**
+ * Calls `platformApi.prepare` for each platform in project
+ *
+ * @param   {string[]}  platformList  List of platforms, added to current project
+ * @param   {string}    projectRoot   Project root directory
+ *
+ * @return  {Promise}
+ */
+function preparePlatforms (platformList, projectRoot) {
+    return Q.all(platformList.map(function(platform) {
+        // TODO: this need to be replaced by real projectInfo
+        // instance for current project.
+        var project = {
+            root: projectRoot,
+            projectConfig: new ConfigParser(cordova_util.projectConfig(projectRoot)),
+            locations: {
+                plugins: path.join(projectRoot, 'plugins'),
+                www: cordova_util.projectWww(projectRoot)
+            }
+        };
+
+        return platforms.getPlatformApi(platform).prepare(project);
+    }));
+}
+
```

Comment:
Yup, factored this to separate function. Another thing that i've missed, is that we still need to do a `browserify` if corresponding option is specified.